haystack/nodes/answer_generator/openai.py

```diff
@@ -53,8 +53,11 @@ class OpenAIAnswerGenerator(BaseGenerator):
     def __init__(
         self,
         api_key: str,
+        base_url: Optional[str] = None,
+        deployment_name: Optional[str] = None,
         model: str = "text-davinci-003",
         max_tokens: int = 50,
+        api_version: str = "2022-12-01",
```

Comment:
I think the docstring for this is missing.

Also, could we add the `Optional` flag here since this isn't required if a user opts for the OpenAI service. 
```python
api_version: Optional[str] = "2022-12-01"
```